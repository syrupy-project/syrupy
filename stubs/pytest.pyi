from typing import (
    Any,
    Callable,
    TypeVar,
)

ReturnType = TypeVar("ReturnType")

def fixture(func: Callable[..., ReturnType]) -> Callable[..., ReturnType]: ...

class Function: ...
class Item: ...
