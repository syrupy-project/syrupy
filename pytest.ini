[pytest]
addopts = -p pytester -p syrupy
