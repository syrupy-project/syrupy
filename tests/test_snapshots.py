from collections import namedtuple
import pytest


def test_simple_string(snapshot):
    assert "Loreeeeeem ipsum." == snapshot


def test_raw_string(snapshot):
    assert r"Raw string" == snapshot


def test_unicode_string(snapshot):
    assert "ü•ûüêçüçØ" == snapshot


def test_multiple_snapshots(snapshot):
    assert "First." == snapshot
    snapshot.assert_match("Second.")
    assert "Third." == snapshot


@pytest.mark.parametrize("expected", [r"Escaped \n", r"Backslash \u U"])
def test_parametrized_with_special_char(snapshot, expected):
    assert expected == snapshot


@pytest.mark.parametrize(
    "actual",
    [
        {"b": True, "c": "Some text.", "d": ["1", 2], "a": {"e": False}},
        {"b": True, "c": "Some ttext.", "d": ["1", 2], "a": {"e": False}},
    ],
)
def test_dict(snapshot, actual):
    assert actual == snapshot


def test_set(snapshot):
    assert snapshot == {"this", "is", "a", "set"}
    assert snapshot == {"this", "is", "a", frozenset({"nested, set"})}


ExampleTuple = namedtuple("ExampleTuple", ["a", "b", "c", "d"])


def test_tuples(snapshot):
    assert snapshot == ("this", "is", ("a", "tuple"))
    assert snapshot == ExampleTuple(a="this", b="is", c="a", d={"named", "tuple"})
