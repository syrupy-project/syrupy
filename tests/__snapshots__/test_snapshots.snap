# name: test_string[0]
    Raw string
    
---
# name: test_string[1]
    singleline:
---
